Document defining query breaking change (#2625)
dotnet/EntityFramework.Docs: entity-framework/core/what-is-new/ef-core-5.0/breaking-changes.md

**Why**

This representation of byte[] aligns better with expectations and is the default behavior of the major JSON serialization libraries.

**Mitigations**

> Existing data stored as number arrays will still be queried correctly, but currently there isn't a supported way to change back the insert behavior. If this change is blocking your scenario, comment on [this issue](https://github.com/aspnet/EntityFrameworkCore/issues/17306)

Reviewer: ```suggestion
Existing data stored as number arrays will still be queried correctly, but currently there isn't a supported way to change back the insert behavior. If this limitation is blocking your scenario, comment on [this issue](https://github.com/aspnet/EntityFrameworkCore/issues/17306)
```

    .Property<string>("Discriminator")
    .Metadata.SetAfterSaveBehavior(PropertySaveBehavior.Save);
```

<a name="defining-query"></a>

### Defining query is replaced with provider specific methods

> [Tracking Issue #21154](https://github.com/dotnet/efcore/issues/18903)

Reviewer: mismatch

**Old behavior**

> Defining queries were defined for entity types at core level. Any time the entity type was used in the query the query root of the entity type was replaced by the defining query for any provider.

Reviewer: ```suggestion
Entity types were mapped to defining queries at the Core level. Any time the entity type was used in the query the query root of the entity type was replaced by the defining query for any provider.
```

**New behavior**

> APIs for defining query are deprecated. Users should use alternate API based on the provider they are using.

Reviewer: ```suggestion
APIs for defining query are deprecated. New provider-specific API were introduced.
```

**Why**

> While defining queries were implemented as replacement expression tree whenever query root is used in the query, it had few issues.

Reviewer: ```suggestion
While defining queries were implemented as replacement expression tree whenever query root is used in the query, it had a few issues:
```

> - If defining query were projecting entity type using `new { ... }` in `Select` method then identifying that has entity type required additional work and made it inconsistent how EF Core treats nominal types in the query.

Reviewer: ```suggestion
- If defining query were projecting entity type using `new { ... }` in `Select` method then identifying that as an entity required additional work and made it inconsistent with how EF Core treats nominal types in the query.
```

> - For relational database users had to use `FromSql` just to pass the SQL string in LINQ expression form.

Reviewer: ```suggestion
- For relational providers `FromSql` was still needed to pass the SQL string in LINQ expression form.
```

> Overall, defining queries were supposed to be client side views to be used with In-Memory provider for keyless entities (similar to database views in relational databases). Such definition makes it easy to test application against in-memory database. It became broad which was useful but brought inconsistent and hard to understand behavior. So we decided to simplify the concept and treat LINQ based defining query exclusive to In-Memory provider and treat it differently. For more information (see this issue) [https://github.com/dotnet/efcore/issues/20023].

Reviewer: ```suggestion
Initially defining queries were introduced as client side views to be used with In-Memory provider for keyless entities (similar to database views in relational databases). Such definition makes it easy to test application against in-memory database. Afterwards they became broadly applicable which was useful but brought inconsistent and hard to understand behavior. So we decided to simplify the concept and make LINQ based defining query exclusive to In-Memory provider and treat them differently. For more information (see this issue) [https://github.com/dotnet/efcore/issues/20023].
```

**Mitigations**

> For relational database, users should use `ToSqlQuery` method in `OnModelCreating` by passing a SQL string to use for the entity type.
> For In-Memory database, users should use `ToInMemoryQuery` method in `OnModelCreating` by passing a LINQ query to use for the entity type.

Reviewer: ```suggestion
For relational providers use `ToSqlQuery` method in `OnModelCreating` and pass in a SQL string to use for the entity type.
For the In-Memory provider use `ToInMemoryQuery` method in `OnModelCreating` and pass in a LINQ query to use for the entity type.
```
